app/policies/project_policy.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+# Policy for projects authorization
+class ProjectPolicy < ApplicationPolicy
+  alias_rule :create?, :edit?, :update?, :new?,
+             to: :allowed_to_modify_project?
+  alias_rule :index?, :show?, :activity?, to: :allowed_to_view_project?
+  alias_rule :destroy?, to: :allowed_to_destroy?
+  alias_rule :transfer?, to: :allowed_to_transfer?
+
+  def allowed_to_view_project?
+    return true if record.namespace.parent.owner == user
+
+    can_view?(record.namespace)
+  end
+
+  def allowed_to_modify_project?
+    return true if record.namespace.parent.owner == user
+
+    can_modify?(record.namespace)
+  end
+
+  def allowed_to_destroy?
+    return true if record.namespace.parent.owner == user
+
+    can_destroy?(record.namespace)
+  end
+
+  def allowed_to_transfer?
+    return true if record.creator == user
```

Comment:
I don't think we want to allow `record.creator` to perform actions, since they should be a member in the namespace or owner of the namespace. In all actuality we should not even be using `owner` in the namespace except for UserNamespace as we have no way of changing that. I,e, we cannot remove an owner once set. Same for creator I could see a scenario where a user created a project in a group but then were removed from the group. This would still allow them to see/modify the project.

Reply:
I think I now got it working as expected. Updated in 04ee06c70a3fd48b1ee58be4f1bf88ff1b28a99d